ts/main.ts

```diff
@@ -0,0 +1,170 @@
+/* exported data */
+interface Data {
+  id: string;
+  image: string;
+  location: string[];
+  title: string;
+  originalTitle: string;
+  originalTitleRomanized: string;
+  description: string;
+  director: string;
+  producer: string;
+  url: string;
+  releaseDate: number;
+  runningTime: number;
+  rtScore: number;
+}
+
+interface Info {
+  data: Data[];
+}
+
+async function filmsData(): Promise<void> {
+  try {
+    const response = await fetch('https://ghibliapi.vercel.app/films');
+    if (!response.ok) {
+      throw new Error(`http error! Status: ${response.status}`);
+    }
+    const films: Data[] = await response.json();
+    const filmographyContainer = document.querySelector(
+      '.filmography-container',
+    ) as HTMLElement;
+    if (!filmographyContainer) {
+      throw new Error('filmography container not found');
+    }
+    filmographyContainer.innerHTML = '';
+
+    films.forEach((film) => {
+      const filmPoster = document.createElement('img');
+      filmPoster.src = film.image;
+      filmPoster.alt = `Poster of ${film.title}`;
+      filmPoster.classList.add('film-poster');
```

Comment:
You should put a `$` before the variable name since you're storing a DOM element